rmw_opendds_cpp/src/rmw_subscription.cpp

```diff
@@ -167,25 +167,21 @@ rmw_create_subscription(
   //  goto fail;
   //}
 
-  if (!topic_name || strlen(topic_name) == 0) {
+  std::string topic_str = get_topic_str(topic_name, qos_profile->avoid_ros_namespace_conventions);
+  if (topic_str.empty()) {
```

Comment:
That's true. It only tells the caller that there is no valid topic_name, and that should be sufficient. Otherwise, we have to pass string& to get the decorated topic_name and return an error code or bool, and the caller has to check the return code before using the value in the string&.